lib/record_store/provider/dnsimple.rb

```diff
@@ -29,76 +22,75 @@ def add(record, zone)
+        session.zones.all_records(account_id, zone).data.map do |record|
           begin
-            build_from_api(record_body, zone)
+            build_from_api(record, zone)
           rescue StandardError
-            stdout.puts "Cannot build record: #{record_body}"
+            stdout.puts "Cannot build record: #{record}"
             raise
           end
         end.compact
       end
 
       # Returns an array of the zones managed by provider as strings
       def zones
-        session.zones.map(&:domain)
+        session.zones.all_zones(account_id).data.map(&:name)
       end
 
       private
 
       def session
-        @dns ||= Fog::DNS.new(session_params)
+        @dns ||= Dnsimple::Client.new(
+          base_url: secrets.fetch('base_url'),
+          access_token: secrets.fetch('api_token')
+        )
       end
 
-      def session_params
-        {
-          provider: 'DNSimple',
-          dnsimple_token: secrets.fetch('api_token'),
-          dnsimple_account: secrets.fetch('account_id'),
-        }
+      def account_id
+        @account_id ||= secrets.fetch('account_id')
       end
 
       def secrets
         super.fetch('dnsimple')
       end
 
       def build_from_api(api_record, zone)
-        record_type = api_record.fetch('type')
+        record_type = api_record.type
         record = {
-          record_id: api_record.fetch('id'),
-          ttl: api_record.fetch('ttl'),
-          fqdn: api_record.fetch('name').present? ? "#{api_record.fetch('name')}.#{zone}" : zone,
+          record_id: api_record.id,
+          ttl: api_record.ttl,
+          fqdn: api_record.name.present? ? "#{api_record.name}.#{zone}" : zone,
         }
 
         return if record_type == 'SOA'
 
         case record_type
         when 'A', 'AAAA'
-          record.merge!(address: api_record.fetch('content'))
+          record.merge!(address: api_record.content)
         when 'ALIAS'
-          record.merge!(alias: api_record.fetch('content'))
+          record.merge!(alias: api_record.content)
         when 'CNAME'
-          record.merge!(cname: api_record.fetch('content'))
+          record.merge!(cname: api_record.content)
         when 'MX'
-          record.merge!(preference: api_record.fetch('priority'), exchange: api_record.fetch('content'))
+          record.merge!(preference: api_record.priority, exchange: api_record.content)
         when 'NS'
-          record.merge!(nsdname: api_record.fetch('content'))
+          record.merge!(nsdname: api_record.content)
         when 'SPF', 'TXT'
-          record.merge!(txtdata: api_record.fetch('content').gsub(';', '\;'))
+          record.merge!(txtdata: api_record.content.gsub(';', '\;'))
         when 'SRV'
-          weight, port, host = api_record.fetch('content').split(' ')
+          weight, port, host = api_record.content.split(' ')
 
           record.merge!(
-            priority: api_record.fetch('priority').to_i,
+            priority: api_record.priority.to_i,
```

Comment:
FYI if this is coming from the API, `priority` should be an integer. We normalized it to be always an integer. But it doesn't hurt to keep `to_i`, I'm not sure if it was in response of some specific issue in the past.

Reply:
Cool! I tried to make as less changes as possible. Will update accordingly now. Thanks for pointing it out. 👍